jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/client/failover/impl/FailoverClosureImpl.java

```diff
@@ -91,7 +92,12 @@ public void success(final T result) {
 
     @Override
     public void failure(final Throwable cause) {
-        this.future.completeExceptionally(cause);
+        if (cause instanceof RemotingException) {
```

Comment:
`RemotingException` belongs to the `bolt` framework and seems not to consider exceptions from other RPC frameworks (although currently rheakkv only supports `bolt`). I think we should avoid introducing specific RPC implementation types in `FailoverClosureImpl`. How about we introduce a `ConnectionFailureException` ?